### PR TITLE
Make Undo wait for animations

### LINKED ISSUE #57: Crash on undo
Sometimes pressing `Undo` crashes the game.
Strack trace shows `src/Undo.lua:48`

### PR DESCRIPTION
**Pull request by Sasszem**
Fixes #57

### CODE REVIEW

**Review by Btimes2 — COMMENTED** [review]
I modded my master branch with this commit, and it works nicely.
The code is also clear.
However, I have same questions before I can merge these changes to the master.
Forgive me, if these are stupid questions.

1. is TasksManager really safe to use? is it another program, or is it just a module of Löve or Yalg?
2. is yield necessary? I am worried that yield could cause high CPU usage on older mobile phones. Or is it smarter than that?
3. also, I began to think that clicking on the Undo button during animation should be simply ignored, instead of waiting for the animation to finish and then quickly do an undo. I think ignoring this would provide a smoother game experience. If I click on it now, then the Undo is so fast that it is hard to follow what happens on the screen. And I think the Undo will be used rarely, so lots of undo clicks will be accidental, too. - I also have an idea about an Undo graphic effect.

**Comment by Sasszem** [comment]
For TaskManager, see "src/TaskManager.lua".
It uses cooperative multitasking. Why would it cause high CPU useage?
Also, all the anumations, the time counter, and even the game end checking uses coroutines!

**Review by Btimes2 — APPROVED** [review]
I will merge this commit to prevent crashes. 
But I will also add an issue about the behavioral changes.
